### ISSUE TITLE
Creating Spark Dataset throws scala.MatchError

### ISSUE THREAD

**Issue by murphyric** [cerner/bunsen#2]
Hi, I get this error while creating a Dataset with a FHIR type resource. Requesting your help.

Spark 2.2
FHIR DSTU3
bunsen-core-0.1.0
PatientSummary is a custom  FHIR Resource

```
Dataset<PatientSummary> patientSummaryDS = sparkSession.createDataset(patientSummaryRDD.collect(),
				encoders.of(PatientSummary.class));
```
Error:
scala.MatchError: RuntimeElementDirectResource[DirectChildResource, IBaseResource] (of class ca.uhn.fhir.context.RuntimeElementDirectResource)
	at com.cerner.bunsen.SchemaConverter.com$cerner$bunsen$SchemaConverter$$childToFields(SchemaConverter.scala:119)
	at com.cerner.bunsen.SchemaConverter$$anonfun$1.apply(SchemaConverter.scala:161)
	at com.cerner.bunsen.SchemaConverter$$anonfun$1.apply(SchemaConverter.scala:161)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.TraversableLike$$anonfun$flatMap$1.apply(TraversableLike.scala:241)
	at scala.collection.Iterator$class.foreach(Iterator.scala:893)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1336)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike$class.flatMap(TraversableLike.scala:241)
	at scala.collection.AbstractTraversable.flatMap(Traversable.scala:104)
	at com.cerner.bunsen.SchemaConverter.compositeToStructType(SchemaConverter.scala:161)
	at com.cerner.bunsen.EncoderBuilder$.of(EncoderBuilder.scala:43)
	at com.cerner.bunsen.EncoderBuilder.of(EncoderBuilder.scala)
	at com.cerner.bunsen.FhirEncoders.of(FhirEncoders.java:64)

**Comment by rbrush** [comment]
Bunsen 0.1.0 supports Spark 2.1.1, and Spark 2.2 unfortunately made some breaking changes to one of its APIs. However, we're about to jump to Spark 2.2 as well, so we'll push a Bunsen branch with 2.2 support probably within the next week.

**Comment by rbrush** [comment]
I pushed Spark 2.2 support into the branch here: https://github.com/cerner/bunsen/tree/issue-2. We'll want to test it on a real cluster before merging, but test pass against a local environment. 

We'll plan on supporting Spark 2.2 in the Bunsen 0.2.x stream and any fixes for Spark 2.1 can live in Bunsen 0.1.x. We'll probably move on from Spark 2.1 entirely before long, but it's useful in our current environment so we started there.

**Comment by rbrush** [comment]
Spark 2.2 support looks to be working well so I merged it to master and pushed a 0.2.0-SNAPSHOT build to the SonaType snapshot repository (https://oss.sonatype.org/content/repositories/snapshots). 

If you're interested in trying Bunsen with Spark 2.2 feel free to point your project at 0.2.0-SNAPSHOT there. Let me know if you continue to see issues with that build.

**Comment by rbrush** [comment]
Bunsen 0.2.2 has been released with Spark 2.2 support and is working well at least in our environment. I'm going to go ahead and close this since that should address the issue, but we can reopen if others experience issues.

**Comment by anam-maker** [comment]
I'm getting the same error when trying to do extract_entry(spark, bundles, 'questionnaireresponse'). I'm using Spark v2.3.0 in bunsen-tutorial. What should I do?
